src/libraries/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -143,7 +143,12 @@ public static NameValueCollection ParseQueryString(string query, Encoding encodi
 
         [return: NotNullIfNotNull(nameof(value))]
         public static string? HtmlEncode(object? value) =>
-            value == null ? null : HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture) ?? string.Empty);
+            value switch
+            {
+                null => null,
+                IHtmlString ihs => ihs.ToHtmlString() ?? string.Empty,
```

Comment:
Nit: the Framework implementation [doesn't alter](https://referencesource.microsoft.com/#System.Web/httpserverutility.cs,1119) `null` to `string.Empty`.

Reply:
I went back and forth on it... part of the problem is since .NET Framework we've annotated the method as returning non-null.

Also, even for the non-IHtmlString case, Framework allows null to be returned and core doesn't, right?  I'd need to double check but I believe that's what I saw.  In which case this is just staying consistent with what the API is doing for other inputs.

Reply:
> Also, even for the non-IHtmlString case, Framework allows null to be returned and core doesn't, right?

Yes - a class that returns a null value from `.ToString()` returns a null value on framework, but a string.Empty on core.

Reply:
Also - should `IHtmlstring.ToHtmlString()` be nullable since we're checking it here?

Reply:
You tell me :-) My assumption was we don't expect null to be valid; the check is just in case. But if null is valid, then yes, this should be nullable. In am the implementations you've seen, it's null expected?

Reply:
Sorry - I'm out sick and slow responding :)  I found a number of implementations in the wild and there are no checks to validate if it is null or not, and the equivalent type on ASP.NET Core accepts a nullable string. Seems like it should be nullable.

Reply:
I don't have a strong opinion here, though, as the type itself is the important thing.

Reply:
I've changed it in https://github.com/dotnet/runtime/pull/85742.